ShortestPath/ShortestPath/ArrayHeap.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ShortestPath
+{
+    public class ArrayHeap : IHeap
+    {
+        public List<Node> Heap = new List<Node>();
+        private int count;
+
+        public Node Root => Heap[0];
+
+        public ArrayHeap(Dictionary<string, Node> nodes)
+        {
+            count = 0;
+            foreach (KeyValuePair<string, Node> node in nodes)
+            {
+                Add(node.Value);
+            }
+        }
+
+        public ArrayHeap()
+        {
+        }
+
+        public void Add(Node node)
+        {
+            Heap.Add(node);
+            count++;
+            PerculateUp();
+        }
+
+        private void PerculateUp()
+        {
+            if (Heap.Count == 0) { return; }
+
+            else
+            {
+                int pointer = count - 1;
+                while (true)
+                {
+                    Node Child = Heap[pointer];
+                    Node Parent = Heap[Math.Abs((pointer) / 2)]; //Maybe need to round down 
+                    Node comparer;
+                    if (Child.Cost < Parent.Cost)
+                    {
+                        comparer = Child;
+                        Heap[pointer] = Heap[Math.Abs((pointer) / 2)];
+                        Heap[Math.Abs((pointer) / 2)] = comparer;
```

Comment:
because each nodes children are :
2n+1
2n+2
while each nodes parent is : 
| (n-1)/2 |  
why abstraction of  (n-1)/2 ? because if our node number is " 3 ", (3-1)/2 = 1.5 and we dont have node by this index in our list so we have to round it, that is why I used abstraction to get nodes parent index.